src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -299,6 +326,66 @@ def enable_zip_deploy(cmd, resource_group_name, name, src, timeout=None, slot=No
     return response
 
 
+def upload_zip_to_storage(cmd, resource_group_name, name, src, slot=None):
+    settings = get_app_settings(cmd, resource_group_name, name, slot)
+
+    storage_connection = None
+    for keyval in settings:
+        if keyval['name'] == 'AzureWebJobsStorage':
+            storage_connection = str(keyval['value'])
+
+    if storage_connection is None:
+        raise CLIError('Could not find a \'AzureWebJobsStorage\' application setting')
+
+    container_name = "function-releases"
+    blob_name = "{}-{}.zip".format(datetime.datetime.today().strftime('%Y%m%d%H%M%S'), str(uuid.uuid4()))
+
+    block_blob_service = BlockBlobService(connection_string=storage_connection)
+    if not block_blob_service.exists(container_name):
+        block_blob_service.create_container(container_name)
+
+    # https://gist.github.com/vladignatyev/06860ec2040cb497f0f3
+    def progress_callback(current, total):
+        total_length = 30
+        filled_length = int(round(total_length * current) / float(total))
+        percents = round(100.0 * current / float(total), 1)
+        progress_bar = '=' * filled_length + '-' * (total_length - filled_length)
+        print('\r{} {}%'.format(progress_bar, percents), flush=True, end="")
+
+    print('Uploading...')
+    block_blob_service.create_blob_from_path(container_name, blob_name, src, validate_content=True,
+                                             progress_callback=progress_callback)
+    # For a new line
+    print('')
+
+    now = datetime.datetime.now()
+    blob_start = now - datetime.timedelta(minutes=10)
+    blob_end = now + datetime.timedelta(weeks=520)
+    blob_token = block_blob_service.generate_blob_shared_access_signature(container_name,
+                                                                          blob_name,
+                                                                          permission=BlobPermissions(read=True),
+                                                                          expiry=blob_end,
+                                                                          start=blob_start)
+
+    blob_uri = block_blob_service.make_blob_url(container_name, blob_name, sas_token=blob_token)
+    website_run_from_setting = "WEBSITE_RUN_FROM_PACKAGE={}".format(blob_uri)
+    print('Upload to storage complete.')
+    update_app_settings(cmd, resource_group_name, name, settings=[website_run_from_setting])
+    print('Syncing triggers...')
+    client = web_client_factory(cmd.cli_ctx)
+
+    try:
+        if slot is not None:
+            client.web_apps.sync_function_triggers_slot(resource_group_name, name, slot)
+        else:
+            client.web_apps.sync_function_triggers(resource_group_name, name)
+    except CloudError as ce:
+        # This SDK function throws an error if Status Code is 200
+        if ce.status_code != 200:
+            raise ce
+    print('Deployment completed successfully.')
```

Comment:
CLI goes with the principle that `silence is golden`. So you only need to raise the error if things went wrong; otherwise just exit silently or return json data if there is

Reply:
@yugangw-msft, that's good to know. I was trying to match the upload behavior exactly with `azure-functions-core-tools`, that is currently used for uploading functions (keeping the messages identical). But, you are right, I will update the PR to keep it silent.